test/cadet_web/controllers/auth_controller_test.exs

```diff
@@ -1,89 +1,90 @@
     AuthController.swagger_path_create(nil)
     AuthController.swagger_path_refresh(nil)
     AuthController.swagger_path_logout(nil)
   end
 
-  describe "POST /v1/auth" do
-    test "missing parameter", %{conn: conn} do
-      conn = post(conn, "/v1/auth", %{})
+  describe "POST /auth" do
+    test "success", %{conn: conn} do
+      use_cassette "auth_controller/v1/auth#1" do
+        conn =
+          post(conn, "/v1/auth", %{
+            "login" => %{"ivle_token" => @token}
+          })
 
-      assert response(conn, 400)
+        assert response(conn, 200)
+      end
     end
 
-    test "blank email", %{conn: conn} do
-      conn =
-        post(conn, "/v1/auth", %{
-          "login" => %{"email" => "", "password" => "password"}
-        })
+    test "missing parameter", %{conn: conn} do
+      conn = post(conn, "/v1/auth", %{})
 
       assert response(conn, 400)
     end
 
-    test "blank password", %{conn: conn} do
+    test "blank token", %{conn: conn} do
       conn =
         post(conn, "/v1/auth", %{
-          "login" => %{"email" => "test@test.com", "password" => ""}
+          "login" => %{"ivle_token" => ""}
         })
 
       assert response(conn, 400)
     end
 
-    test "email not found", %{conn: conn} do
-      conn =
-        post(conn, "/v1/auth", %{
-          "login" => %{"email" => "unknown@test.com", "password" => "password"}
-        })
-
-      assert response(conn, 403)
-    end
-
-    test "invalid password", %{conn: conn} do
-      test_password = "password"
-      user = insert(:user)
-
-      email =
-        insert(:email, %{
-          token: Pbkdf2.hash_pwd_salt(test_password),
-          user: user
-        })
-
-      conn =
-        post(conn, "/v1/auth", %{
-          "login" => %{"email" => email.uid, "password" => "password2"}
-        })
-
-      assert response(conn, 403)
+    test "invalid token", %{conn: conn} do
+      use_cassette "auth_controller/v1/auth#2" do
+        conn =
+          post(conn, "/v1/auth", %{
+            "login" => %{"ivle_token" => @token <> "Z"}
+          })
+
+        assert response(conn, 400)
+      end
     end
 
-    test "valid email and password", %{conn: conn} do
-      test_password = "password"
-      user = insert(:user)
-
-      email =
-        insert(:email, %{
-          token: Pbkdf2.hash_pwd_salt(test_password),
-          user: user
-        })
-
-      conn =
-        post(conn, "/v1/auth", %{
-          "login" => %{
-            "email" => email.uid,
-            "password" => test_password
-          }
-        })
-
-      assert json_response(conn, 200)
+    test "invalid nusnet id", %{conn: conn} do
+      # an invalid nusnet id == ~s("") is typically caught by IVLE.fetch_nusnet_id
+      # the custom cassette skips this step so that we can test Accounts.sign_in
+      use_cassette "auth_controller/v1/auth#1", custom: true do
+        conn =
+          post(conn, "/v1/auth", %{
+            "login" => %{"ivle_token" => 'token'}
```

Comment:
Note that in elixir `'charlist'` and `"binary"` are 2 different ways of storing strings https://elixir-lang.org/getting-started/binaries-strings-and-char-lists.html

it should be `"token"` instead.

More to the point, it still doesn't cover: reading the report generated from running `mix coveralls.html test/cadet_web/controllers/auth_controller_test.exs:77`, the error is "caught" at `{:fetch, {:ok, nusnet_id}} <- {:fetch, IVLE.fetch_nusnet_id(login.ivle_token)}` since `api_fetch/2` returns `{:error, :bad_request}` on `%{body: ~s(""), status_code: 200}`.

One way to solve this might be to handle `{:error, _}` in `api_fetch/2`, then purposely fail the request `IVLE.fetch_name/1` if that makes sense.

Reply:
My bad, I should've checked the coveralls link. Fixed with 59a315f using custom cassettes. The fix didn't need the catch "don't care" clause, but I'll leave that change in.